native/dlaqr2.go

```diff
@@ -0,0 +1,382 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/lapack"
+)
+
+// Dlaqr2 performs the orthogonal similarity transformation of a n×n upper
+// Hessenberg matrix to detect and deflate fully converged eigenvalues from a
+// trailing principal submatrix using aggressive early deflation [1].
+//
+// On return, H will be overwritten by a new Hessenberg matrix that is a
+// perturbation of an orthogonal similarity transformation of H. It is to be
+// hoped that on output H will have many zero subdiagonal entries.
+//
+// If wantt is true, the matrix H will be fully updated so that the
+// quasi-triangular Schur factor can be computed. If wantt is false, then only
+// enough of H will be updated to preserve the eigenvalues.
+//
+// If wantz is true, the orthogonal similarity transformation will be
+// accumulated into Z[iloz:ihiz+1,ktop:kbot+1], otherwise Z is not referenced.
+//
+// ktop and kbot determine an block [ktop:kbot+1,ktop:kbot+1] along the diagonal
+// of H. The block must be isolated, that is, it must hold that
+//  ktop == 0   or H[ktop,ktop-1] == 0,  and
+//  kbot == n-1 or H[kbot+1,kbot] == 0,
+// otherwise Dlaqr2 will panic.
+//
+// nw is the deflation window size. It must hold that
+//  1 <= nw <= kbot-ktop+1,
+// otherwise Dlaqr2 will panic.
+//
+// iloz and ihiz specify the rows of the n×n matrix Z to which transformations
+// will be applied if wantz is true. It must hold that
+//  0 <= iloz <= ktop,  and  kbot <= ihiz < n,
+// otherwise Dlaqr2 will panic.
+//
+// sr and si must have length kbot+1, otherwise Dlaqr2 will panic.
+//
+// v and ldv represent an nw×nw work matrix.
+//
+// t and ldt represent an nw×nw work matrix T. nh is the number of columns of T
+// available for workspace. It must hold that nh <= nw, otherwise Dlaqr2 will
+// panic.
+//
+// wv and ldwv represent an nw×nw work matrix W. nv is the number of rows of W
+// available for workspace. It must hold that nv <= nw, otherwise Dlaqr2 will
+// panic.
+//
+// work must have length at least lwork and lwork must be at least max(1,2*nw),
+// otherwise Dlaqr2 will panic. Larger values of lwork may result in greater
+// efficiency. On return, work[0] will contain the optimal value of lwork.
```

Comment:
Yes, they do (not sure why). Inexhaustive search gave me at least Dgehrd, Dorghr, Dormhr, Dorgbr.